cf/commands/domain/create_shared_domain_test.go

```diff
@@ -1,73 +1,308 @@
+					flagContext.Parse("domain-name", "--router-group", "route-group-name")
+				})
+
+				It("returns a LoginRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(factory.NewLoginRequirementCallCount()).To(Equal(1))
+					Expect(actualRequirements).To(ContainElement(loginRequirement))
+				})
+
+				It("returns a RoutingApiRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(factory.NewRoutingAPIRequirementCallCount()).To(Equal(1))
+					Expect(actualRequirements).To(ContainElement(routingApiRequirement))
+				})
+
+				It("returns a MinAPIVersionRequirement", func() {
+					expectedVersion, err := semver.Make("2.36.0")
+					Expect(err).NotTo(HaveOccurred())
 
-		runCommand("example.com")
-		Expect(ui.Outputs).ToNot(ContainSubstrings(
-			[]string{"Incorrect Usage", "Requires an argument"},
-		))
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(factory.NewMinAPIVersionRequirementCallCount()).To(Equal(1))
+					feature, requiredVersion := factory.NewMinAPIVersionRequirementArgsForCall(0)
+					Expect(feature).To(Equal("Option '--router-group'"))
+					Expect(requiredVersion).To(Equal(expectedVersion))
+					Expect(actualRequirements).To(ContainElement(minAPIVersionRequirement))
+				})
+			})
+
+			Context("when router-group flag is not set", func() {
+				BeforeEach(func() {
+					flagContext.Parse("domain-name")
+				})
+
+				It("returns a LoginRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(factory.NewLoginRequirementCallCount()).To(Equal(1))
+					Expect(actualRequirements).To(ContainElement(loginRequirement))
+				})
+
+				It("does not return a RoutingApiRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(factory.NewRoutingAPIRequirementCallCount()).To(Equal(0))
+					Expect(actualRequirements).ToNot(ContainElement(routingApiRequirement))
+				})
+
+				It("does not return a MinAPIVersionRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(actualRequirements).NotTo(ContainElement(minAPIVersionRequirement))
+				})
+			})
+		})
 	})
 
-	It("TestShareDomain", func() {
-		runCommand("example.com")
+	Describe("Execute", func() {
+		Context("when router-group flag is not set", func() {
+			BeforeEach(func() {
+				flagContext.Parse("domain-name")
+				cmd.Execute(flagContext)
+			})
+
+			It("prints a message", func() {
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Creating shared domain domain-name"},
+				))
+			})
+
+			It("creates a shared domain", func() {
```

Comment:
This should say `it tries to create a shared domain`; further contexts below should describe what happens when the call succeeds or fails.